core/src/labels/labels.cpp

```diff
@@ -107,6 +106,65 @@ void Labels::update(const View& _view, float _dt, const std::vector<std::unique_
     }
 }
 
+const std::vector<std::shared_ptr<Properties>>& Labels::getFeaturesAtPoint(const View& _view, float _dt,
+                                                                           const std::vector<std::unique_ptr<Style>>& _styles,
+                                                                           const std::vector<std::shared_ptr<Tile>>& _tiles,
+                                                                           float _x, float _y, bool _visibleOnly) {
+    // FIXME dpi dependent threshold
+    const float thumbSize = 20;
+    const float threshold = 100;
+
+    m_touchItems.clear();
+
+    glm::vec2 screenSize = glm::vec2(_view.getWidth(), _view.getHeight());
+    glm::vec2 touchPoint(_x, _y);
+
+    OBB obb(_x - thumbSize/2, _y - thumbSize/2, 0, thumbSize, thumbSize);
+    m_touchPoint = obb;
+
+    for (const auto& tile : _tiles) {
+
+        glm::mat4 mvp = _view.getViewProjectionMatrix() * tile->getModelMatrix();
+
+        for (const auto& style : _styles) {
+            const auto& mesh = tile->getMesh(*style);
+            if (!mesh) { continue; }
+
+            const LabelMesh* labelMesh = dynamic_cast<const LabelMesh*>(mesh.get());
+            if (!labelMesh) { continue; }
+
+            for (auto& label : labelMesh->getLabels()) {
+
+                auto& options = label->getOptions();
+                if (!options.interactive) { continue; }
+
+                auto& transform = label->getTransform();
+                glm::vec4 v1 = worldToClipSpace(mvp, glm::vec4(transform.modelPosition1, 0.0, 1.0));
+                glm::vec4 v2 = worldToClipSpace(mvp, glm::vec4(transform.modelPosition2, 0.0, 1.0));
+
+                // check whether the label is behind the camera using the perspective division factor
+                if (v1.w <= 0 || v2.w <= 0) { continue; }
+
+                // project to screen space
+                glm::vec2 p1 = clipToScreenSpace(v1, screenSize);
+                glm::vec2 p2 = clipToScreenSpace(v2, screenSize);
+
+                float distance = sqrt(sqSegmentDistance(touchPoint, p1, p2));
+
+                if (distance > threshold) { continue; }
+
+                if (!_visibleOnly || (label->visibleState() && isect2d::intersect(label->getOBB(), obb))) {
```

Comment:
Yeah you're right, I'm not sure what should be the best behavior for those invisible labels, since selecting nothing doesn't mean no feature are present (even when you try on invisible labels); but  as a temporary solution both ways would be fine (distance or bbox) depending on how much precision we want. We can keep this as is for now